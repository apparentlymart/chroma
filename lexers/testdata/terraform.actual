variable "vpc_id" {
  type = string
}

module "us-east-1" {
  source          = "./region"
  region          = "us-east-1"
  base_cidr_block = var.base_cidr_block
}

resource "aws_subnet" "main" {
  cidr_block = cidrsubnet(data.aws_vpc.target.cidr_block, 4, local.az_numbers["b"]))

  lifecycle {
    create_before_destroy = true
  }
}

# Shared security group for allowing HTTP
resource "aws_security_group" "http" {
  name = "${var.vpc_id}-http"

  ingress {
    from_port = 80
    to_port   = 80
  }
}
