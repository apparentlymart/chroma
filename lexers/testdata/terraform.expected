[
  { "type": "KeywordDeclaration", "value": "variable" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringName", "value": "\"vpc_id\"" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "{" },
  { "type": "Text", "value": "\n  " },
  { "type": "Keyword", "value": "type" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "KeywordType", "value": "string" },
  { "type": "Text", "value": "\n" },
  { "type": "Punctuation", "value": "}" },

  { "type": "Text", "value": "\n\n" },

  { "type": "KeywordDeclaration", "value": "module" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringName", "value": "\"us-east-1\"" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "{" },
  { "type": "Text", "value": "\n  " },
  { "type": "Keyword", "value": "source" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "LiteralString", "value": "./region" },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "Text", "value": "\n  " },
  { "type": "NameAttribute", "value": "region" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "LiteralString", "value": "us-east-1" },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "Text", "value": "\n  " },
  { "type": "NameAttribute", "value": "base_cidr_block" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "NameVariable", "value": "var" },
  { "type": "Punctuation", "value": "." },
  { "type": "NameAttribute", "value": "base_cidr_block" },
  { "type": "Text", "value": "\n" },
  { "type": "Punctuation", "value": "}" },

  { "type": "Text", "value": "\n\n" },

  { "type": "KeywordDeclaration", "value": "resource" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringName", "value": "\"aws_subnet\"" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringName", "value": "\"main\"" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "{" },
  { "type": "Text", "value": "\n  " },
  { "type": "NameAttribute", "value": "cidr_block" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "NameFunction", "value": "cidrsubnet" },
  { "type": "Punctuation", "value": "(" },
  { "type": "NameVariable", "value": "data" },
  { "type": "Punctuation", "value": "." },
  { "type": "NameAttribute", "value": "aws_vpc" },
  { "type": "Punctuation", "value": "." },
  { "type": "NameAttribute", "value": "target" },
  { "type": "Punctuation", "value": "." },
  { "type": "NameAttribute", "value": "cidr_block" },
  { "type": "Punctuation", "value": "," },
  { "type": "Text", "value": " " },
  { "type": "LiteralNumber", "value": "4" },
  { "type": "Punctuation", "value": "," },
  { "type": "Text", "value": " " },
  { "type": "NameVariable", "value": "local" },
  { "type": "Punctuation", "value": "." },
  { "type": "NameAttribute", "value": "az_numbers" },
  { "type": "Punctuation", "value": "[" },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "LiteralString", "value": "b" },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "Punctuation", "value": "]" },
  { "type": "Punctuation", "value": ")" },
  { "type": "Text", "value": "\n\n  " },
  { "type": "Keyword", "value": "lifecycle" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "{" },
  { "type": "Text", "value": "\n    " },
  { "type": "Keyword", "value": "create_before_destroy" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "Literal", "value": "true" },
  { "type": "Text", "value": "\n  " },
  { "type": "Punctuation", "value": "}" },
  { "type": "Text", "value": "\n" },
  { "type": "Punctuation", "value": "}" },

  { "type": "Text", "value": "\n\n" },

  { "type": "KeywordDeclaration", "value": "resource" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringName", "value": "\"aws_security_group\"" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringName", "value": "\"http\"" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "{" },
  { "type": "Text", "value": "\n  " },
  { "type": "NameAttribute", "value": "name" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "LiteralStringInterpol", "value": "${" },
  { "type": "NameVariable", "value": "var" },
  { "type": "Punctuation", "value": "." },
  { "type": "NameAttribute", "value": "vpc_id" },
  { "type": "LiteralStringInterpol", "value": "}" },
  { "type": "LiteralString", "value": "-http" },
  { "type": "LiteralStringDelimiter", "value": "\"" },
  { "type": "Text", "value": "\n\n  " },
  { "type": "NameAttribute", "value": "ingress" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "{" },
  { "type": "Text", "value": "\n  " },
  { "type": "NameAttribute", "value": "from_port" },
  { "type": "Text", "value": " " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "LiteralNumber", "value": "80" },
  { "type": "Text", "value": "\n  " },
  { "type": "NameAttribute", "value": "to_port" },
  { "type": "Text", "value": "   " },
  { "type": "Punctuation", "value": "=" },
  { "type": "Text", "value": " " },
  { "type": "LiteralNumber", "value": "80" },
  { "type": "Text", "value": "\n  " },
  { "type": "Punctuation", "value": "}" },
  { "type": "Text", "value": "\n" },
  { "type": "Punctuation", "value": "}" },

  { "type": "Text", "value": "\n" }
]
